src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java

```diff
@@ -179,22 +182,22 @@ public List<NamedXContentRegistry.Entry> getNamedXContent() {
             );
     }
 
-    // TODO: replace or override client object on BaseExtension
-    // https://github.com/opensearch-project/opensearch-sdk-java/issues/160
-    public OpenSearchClient getClient() {
+    private OpenSearchAsyncClient createJavaAsyncClient(ExtensionsRunner runner) {
         @SuppressWarnings("resource")
-        OpenSearchClient client = new SDKClient()
-            .initializeJavaClient(
-                getExtensionSettings().getOpensearchAddress(),
-                Integer.parseInt(getExtensionSettings().getOpensearchPort())
-            );
+        OpenSearchAsyncClient client = runner.getSdkClient().initializeJavaAsyncClient();
         return client;
     }
 
+    // TODO: replace or override client object on BaseExtension
+    // https://github.com/opensearch-project/opensearch-sdk-java/issues/160
+    public OpenSearchAsyncClient getClient() {
```

Comment:
Since we have the getter for `sdkRestClient` as `restClient`. We can keep this method name as `javaClient`

Reply:
how about `javaAsyncClient()`?